oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/TableMetaInfo.java

```diff
@@ -26,6 +26,8 @@ public class TableMetaInfo {
     private static Map<String, Model> TABLES = new HashMap<>();
 
     public static void addModel(Model model) {
+        // remove exclude columns according to @SQLDatabase.AdditionalEntity
+        model.getColumns().removeAll(model.getSqlDBModelExtension().getExcludeColumns());
```

Comment:
You are replicating this line of code everywhere you need `model.getColumns()`, consider override the constructor to exclude the duplication.


```diff
diff --git a/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/model/Model.java b/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/model/Model.java
index eea3a61f8..e258dfae8 100644
--- a/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/model/Model.java
+++ b/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/model/Model.java
@@ -50,7 +50,6 @@ public class Model {
                  boolean timeRelativeID,
                  final SQLDatabaseModelExtension sqlDBModelExtension) {
         this.name = name;
-        this.columns = columns;
         this.scopeId = scopeId;
         this.downsampling = downsampling;
         this.isTimeSeries = !DownSampling.None.equals(downsampling);
@@ -59,5 +58,9 @@ public class Model {
         this.aggregationFunctionName = aggregationFunctionName;
         this.timeRelativeID = timeRelativeID;
         this.sqlDBModelExtension = sqlDBModelExtension;
+        this.columns = columns;
+        if (sqlDBModelExtension != null) {
+            this.columns.removeAll(sqlDBModelExtension.getExcludeColumns());
+        }
     }
 }

```

Reply:
Inside the constructor would affect other storage, this feature should only work on SQL DB...